src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala

```diff
@@ -125,53 +89,57 @@ Expected DataFrame Row Count: '${expectedCount}'
-    if (!SchemaComparer.equals(actualDS.schema, expectedDS.schema, ignoreNullable, ignoreColumnNames)) {
-      throw DatasetSchemaMismatch(betterSchemaMismatchMessage(actualDS, expectedDS))
-    }
-    // then check if the DataFrames have the same content
-    def throwIfDatasetsAreUnequal(ds1: Dataset[T], ds2: Dataset[T]) = {
-      try {
-        ds1.rdd.cache
-        ds2.rdd.cache
-
-        val actualCount   = ds1.rdd.count
-        val expectedCount = ds2.rdd.count
-
-        if (actualCount != expectedCount) {
-          throw DatasetCountMismatch(countMismatchMessage(actualCount, expectedCount))
-        }
-
-        val expectedIndexValue: RDD[(Long, T)] = RddHelpers.zipWithIndex(ds1.rdd)
-        val resultIndexValue: RDD[(Long, T)]   = RddHelpers.zipWithIndex(ds2.rdd)
-        val unequalRDD = expectedIndexValue
-          .join(resultIndexValue)
-          .filter { case (idx, (o1, o2)) =>
-            !equals(o1, o2)
-          }
-        val maxUnequalRowsToShow = 10
-        if (!unequalRDD.isEmpty()) {
-          throw DatasetContentMismatch(
-            unequalRDDMessage(unequalRDD, maxUnequalRowsToShow)
-          )
-        }
-        unequalRDD.take(maxUnequalRowsToShow)
-
-      } finally {
-        ds1.rdd.unpersist()
-        ds2.rdd.unpersist()
-      }
-    }
+    SchemaComparer.assertSchemaEqual(actualDS, expectedDS, ignoreNullable, ignoreColumnNames, ignoreColumnOrder)
+    val actual = if (ignoreColumnOrder) orderColumns(actualDS, expectedDS) else actualDS
+    assertLargeDatasetContentEquality(actual, expectedDS, equals, orderedComparison)
+  }
 
+  def assertLargeDatasetContentEquality[T: ClassTag](actualDS: Dataset[T],
+                                                     expectedDS: Dataset[T],
+                                                     equals: (T, T) => Boolean,
+                                                     orderedComparison: Boolean): Unit = {
     if (orderedComparison) {
-      throwIfDatasetsAreUnequal(actualDS, expectedDS)
+      assertLargeDatasetContentEquality(actualDS, expectedDS, equals)
     } else {
-      throwIfDatasetsAreUnequal(sortPreciseColumns(actualDS), sortPreciseColumns(expectedDS))
+      assertLargeDatasetContentEquality(sortPreciseColumns(actualDS), sortPreciseColumns(expectedDS), equals)
+    }
+  }
+
+  def assertLargeDatasetContentEquality[T: ClassTag](ds1: Dataset[T], ds2: Dataset[T], equals: (T, T) => Boolean): Unit = {
+    try {
+      val ds1RDD = ds1.rdd.cache()
+      val ds2RDD = ds2.rdd.cache()
+
+      val actualCount   = ds1RDD.count
+      val expectedCount = ds2RDD.count
+
+      if (actualCount != expectedCount) {
+        throw DatasetCountMismatch(countMismatchMessage(actualCount, expectedCount))
+      }
+
+      val unequalRDD = ds1RDD
+        .zip(ds2RDD)
+        .filter { case (o1: T, o2: T) => !equals(o1, o2) }
+        .zipWithIndex()
+        .map { case ((t1, t2), idx) => (idx, (t1, t2)) }
+      val maxUnequalRowsToShow = 10
```

Comment:
If we already changing this part, may we move this constant to the companion object?